scim-core/src/main/java/org/apache/directory/scim/core/repository/PatchHandlerImpl.java

```diff
@@ -104,28 +92,19 @@ private <T extends ScimResource> T apply(final T source, final PatchOperation pa
     if (valuePathExpression.getAttributeExpression() != null && attributeObject instanceof Collection<?>) {
       // apply expression filter
       Collection<Object> items = (Collection<Object>) attributeObject;
-      Collection<Object> updatedCollection = items.stream().map(item -> {
-        KeyedResource keyedResource = (KeyedResource) item;
-        Map<String, Object> keyedResourceAsMap = objectAsMap(item);
-        Predicate<KeyedResource> pred = FilterExpressions.inMemory(valuePathExpression.getAttributeExpression(), baseSchema);
-        if (pred.test(keyedResource)) {
+      items.forEach(item -> {
+        Predicate<Object> pred = FilterExpressions.inMemory(valuePathExpression.getAttributeExpression(), baseSchema);
+        if (pred.test(item)) {
           String subAttributeName = valuePathExpression.getAttributePath().getSubAttributeName();
-          if (keyedResourceAsMap.get(subAttributeName) == null && patchOperation.getOperation().equals(PatchOperation.Type.REPLACE)) {
-            throw new IllegalArgumentException("Cannot apply patch replace on missing property: " + valuePathExpression.toFilter());
-          }
-          keyedResourceAsMap.put(subAttributeName, patchOperation.getValue());
-          return keyedResourceAsMap;
-        } else {
-          // filter does not apply
-          return item;
+          Schema.AttributeAccessor subAttributeAccessor = attribute.getAttribute(subAttributeName).getAccessor();
+          subAttributeAccessor.set(item, patchOperation.getValue());
```

Comment:
I'm not 100% sure this will fit what you had in mind originally, but we should be able to use the accessor `set` method here.
We may need to test cases where the field is immutable, (and make sure the correct exception is thrown), but that _should_ happen at the accessor level (and ideally should happen with the ObjectMapper too).

I'll try to create a test for this (as that's independent of my change here)